pkg/alertmanager/multitenant.go

```diff
@@ -1011,6 +1012,102 @@ func (am *MultitenantAlertmanager) ReplicateStateForUser(ctx context.Context, us
 	return err
 }
 
+// getReplicasForUser returns the addresses for each _other_ replica assigned to a user.
+func (am *MultitenantAlertmanager) getOtherReplicasForUser(userID string) ([]string, error) {
+	if am.ring.InstancesCount() <= 0 {
+		return nil, fmt.Errorf("InstancesCount <= 0")
+	}
+
+	key := shardByUser(userID)
+	replicationSet, err := am.ring.Get(key, RingOp, nil, nil, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	// Don't return our own address in the list.
+	selfAddress := am.ringLifecycler.GetInstanceAddr()
+
+	result := make([]string, 0, len(replicationSet.Instances))
```

Comment:
You can call `replicationSet.GetAddresses()` to get the list of addresses. You can add another function like `ReplicationSet.GetAddressesWithout(exclude ...string)` for your convenience.

Once you apply both comments, this function becomes so small that I believe you don't even need it and you can embed the few lines of code directly in `ReadFullStateForUser()`.